src/fft.rs

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module implements an iterative FFT algorithm for computing the (inverse) Discrete Fourier
+//! Transform (DFT) over a slice of field elements.
+
+use crate::finite_field::FieldElement;
+use crate::util;
+
+/// Sets `outp` to the DFT of `inp`.
+pub fn dft<F: FieldElement>(outp: &mut [F], inp: &[F]) {
```

Comment:
That's right, the goal here is to avoid unnecessary memory allocation.

Reply:
I'm fine with avoiding allocation, but should point out that in Rust, unless you do `unsafe` stuff or use something like [`Box`](https://doc.rust-lang.org/std/boxed/struct.Box.html), you never touch the heap. If you create `[F]` and return it from a function, that all happens on the stack (modulo whatever an implementation of `FieldElement` does). So if the concern is to avoid `malloc(3)` or `brk(2)` then IIUC that wasn't happening anyway. However while they're cheaper than the heap, stack allocations still aren't free, and this pattern fits the existing FFT code so I'm comfortable with it.

Reply:
Ack. Yeah there's a lot I don't know about the internals of Rust here.